clang/test/CIR/CodeGen/delegating-ctor.cpp

```diff
@@ -70,3 +70,266 @@ DelegatingWithZeroing::DelegatingWithZeroing(int) : DelegatingWithZeroing() {}
 // OGCG:   store i32 %[[I_ARG]], ptr %[[I_ADDR]]
 // OGCG:   %[[THIS:.*]] = load ptr, ptr %[[THIS_ADDR]]
 // OGCG:   call void @llvm.memset.p0.i64(ptr align 4 %[[THIS]], i8 0, i64 4, i1 false)
+
+void other();
+
+class Base {
+public:
+  Base() { squawk(); }
+
+  virtual void squawk();
+};
+
+class Derived : public virtual Base {
+public:
+  Derived();
+  Derived(const void *inVoid);
+
+  virtual void squawk();
+};
+
+Derived::Derived() : Derived(nullptr) { other(); }
+Derived::Derived(const void *inVoid) { squawk(); }
+
+// Note: OGCG emits the constructors in a different order.
```

Comment:
Does this just mean the 'define'/'declaration' bits?  Or calls themselves?  Could you use OGCG-DAG to get it out of order/reset right?

Reply:
The entire definitions of the constructors are ordered differently in OGCG. I don't think I can use OGCG-DAG to correct for this because once I've found the start of the definition, I need to make the rest of the checks in order. It makes it difficult/impossible to compare the LLVM/OGCG checks just by looking, but I can tell you that I created the OGCG checks by copying the LLVM checks and then updating them as needed and the differences were expected and minor, mostly (a) extra attributes, such as `noundef` on arguments, or (b) different but equivalent GEP formulas.

Reply:
You SHOULD be able to CHECK-DAG on the 'first' line (the define one?), then all the other searching will happen in the right order?   Might be nice just to have a side-by-side-comparison.

That said, the test probably (and perhaps others) could use judicious use of -NEXT as well.

Reply:
I tried CHECK-DAG, CHECK-DAG with CHECK-NEXT, and CHECK-LABEL. None worked. I found an [old discussion](https://groups.google.com/g/llvm-dev/c/q1TDIf6urDI/m/2Fcny3xADgAJ) where someone asked for CHECK-LABEL-DAG to do exactly this and even had a [patch to implement it](https://reviews.llvm.org/D22348), but I guess it was ultimately judged to be unwanted. ☹️